contrib/AutoGrouper.lua

```
@@ -0,0 +1,155 @@
+	if test >= low and test <= high then
+		return true
+	else
+		return false
+	end
+end
+
+local function CompTime(first, second) --compares the timestamps and returns true if first was taken before second
+	first_time = first.exif_datetime_taken
+	if string.match(first_time, '[0-9]') == nil then first_time = '9999:99:99 99:99:99' end
+	first_time = tonumber(string.gsub(first_time, '[^0-9]*',''))
+	second_time = second.exif_datetime_taken
+	if string.match(second_time, '[0-9]') == nil then second_time = '9999:99:99 99:99:99' end
+	second_time = tonumber(string.gsub(second_time, '[^0-9]*',''))
+	return first_time < second_time
+end
+
+local function SeperateTime(str) --seperates the timestamp into individual components for used with OS.time operations
+	local cleaned = string.gsub(str, '[^%d]',':')
+	cleaned = string.gsub(cleaned, '::*',':')  --YYYY:MM:DD:hh:mm:ss
+	local year = string.sub(cleaned,1,4)
+	local month = string.sub(cleaned,6,7)
+	local day = string.sub(cleaned,9,10)
+	local hour = string.sub(cleaned,12,13)
+	local min = string.sub(cleaned,15,16)
+	local sec = string.sub(cleaned,18,19)
+	return {year = year, month = month, day = day, hour = hour, min = min, sec = sec}
+end
+
+local function GetTimeDiff(curr_image, prev_image) --returns the time difference (in sec.) from current image and the previous image
+	local curr_time = SeperateTime(curr_image.exif_datetime_taken)
+	local prev_time = SeperateTime(prev_image.exif_datetime_taken)
+	return os.time(curr_time)-os.time(prev_time)
+end
+
+local function main(on_collection)
+	local images = {}
+	if on_collection then 
+		local col_images = dt.collection
+		for i,image in ipairs(col_images) do --copy images to a standard table, table.sort barfs on type dt_lua_singleton_image_collection
+			table.insert(images,i,image)
+		end
+	else
+		images = dt.gui.selection()
+	end
+	dt.preferences.write(mod, 'active_gap', 'integer', GUI.gap.value)
+	if #images < 2 then return end

```

Maybe print a message to the screen (and/or log) about insufficient images instead of just silently doing nothing